### PR TITLE
PCMSolver is no longer a git submodule.

### PR DESCRIPTION
**Pull request by robertodr**
We use the strategy devised by @loriab for external add ons. This commit provides the necessary Find* and Config* CMake script files. Other modifications take into account the new strategy.

### CODE REVIEW

**Comment by loriab** [comment]
Yay, thanks! I may actually get the psi4 conda package working again tonight with this in place. It can get a little crazy with 4 repos.

**Comment by jturney** [comment]
Nice, I've got this follow for updating Ambit. :+1:

File: lib/python/pcm_placeholder.py.in [hunk 1]
```diff
@@ -15,4 +15,4 @@ that has to be configured by CMake
 
 """
 
-PCMSOLVER_PARSE_DIR='@PCMSOLVER_PARSE_DIR@'
+PCMSolver_PARSE_DIR='@PCMSolver_PARSE_DIR@'
```
Comment (loriab) [inline lib/python/pcm_placeholder.py.in]: Note that this does break equality of psi4-top-dir/lib and installed/share/psi/lib equality for the first time.

This isn't necessarily bad, as it had to happen some time. (Reasonably, add-ons may want to install into driver or lib.) But it does mean that running from uninstalled objdir with PSIDATADIR isn't equiv to running from installed. Just be aware.

Comment (robertodr) [inline lib/python/pcm_placeholder.py.in]: @loriab yes, this glitch slipped through.
`PCMSolver_PARSE_DIR` is defined as `${INSTALL_DIR}/bin` aka `${PROJECT_BINARY_DIR}/interfaces/pcmsolver/install/bin` if the code gets compiled on the spot. If the library was compiled somewhere else, the value `${PCMSOLVER_ROOT}/bin` is used. In the former case, we should install to the global install prefix. Problem is: the install step always happens for CMake subprojects and the default install prefix requires root privileges...

Comment (loriab) [inline lib/python/pcm_placeholder.py.in]: Yeah, it's a problem with no good sol'n yet. Just good to monitor the use cases.

**Pull request merged**